src/H5D.c

```diff
@@ -295,7 +295,7 @@ H5Dcreate_anon(hid_t loc_id, hid_t type_id, hid_t space_id, hid_t dcpl_id, hid_t
         HGOTO_ERROR(H5E_DATASET, H5E_CANTSET, H5I_INVALID_HID, "can't set access property list info");
 
     /* get the location object */
-    if (NULL == (vol_obj = (H5VL_object_t *)H5I_object(loc_id)))
+    if (NULL == (vol_obj = H5VL_vol_object(loc_id)))
```

Comment:
Is there a reason this isn't another instance of `H5VL_vol_object_verify()`? It seems like the type is expected to be a group or file.

Reply:
This ID is supposed to be a file, group, dataset, named datatype, or attribute.  Verifying multiple object types is another issue we'll need to resolve at another time.  This PR is concentrating on H5I_object() vs H5VL_vol_object().